src/client/user/components/UserLinkTable/ToolBar/SearchInput/index.tsx

```diff
@@ -110,26 +110,28 @@ const textFieldHeight = useSearchInputHeight()
 // Search Input field.
 const SearchInput = () => {
   const dispatch = useDispatch()
-  const [query, setQuery] = useState('')
 
   const tableConfig = useSelector(
     (state: GoGovReduxState) => state.user.tableConfig,
   )
+  const { isTag, searchText } = tableConfig
   const searchInputHeight = useSearchInputHeight()
   const classes = useStyles({ textFieldHeight, searchInputHeight })
 
-  const changeSearchTextHandler = (e: string) => {
-    setQuery(e)
+  const setSearchText = (searchText: string) => {
```

Comment:
Just leaving a general comment here, not specific to this line but the effects in general throughout the file.

It seems a bit untidy that we are dispatching the urlTable config from multiple places throughout the code. Is this necessary?

Seems like the goal is to dispatch if something changes, but implement debouncing so that it's not constantly fetching. Is that right? If so, can we abstract that out somehow?

Reply:
> It seems a bit untidy that we are dispatching the urlTable config from multiple places throughout the code. Is this necessary?

can't see a cleaner way to do this sadly :( any suggestions on how this can be improved?

> Seems like the goal is to dispatch if something changes, but implement debouncing so that it's not constantly fetching. Is that right? If so, can we abstract that out somehow?

indeed, though I'm not aware of any other way to do debouncing apart from using `useEffect` and cleanup functions, and I can't think of where else to place `useEffect` apart from at the component level (e.g. here in this component) - would there be other approaches we can consider?